lib/rest_5.js

```diff
@@ -590,40 +590,61 @@ function* createTestUsers(uid, password = '1234') {
   return { admin, master, attacker }
 }
 
-function* waitQuery(queryString, count, timeoutMilli, node) {
-  if (queryString === undefined) throw new Error('waitQuery: queryString undefined');
-  if (count <= 0 ) throw new Error('waitQuery: illegal count');
+function* until(predicate, action, timeoutMilli, node) {
```

Comment:
I think exponential backoff would make it less punitive to choose a too high timeoutMilli.
```
const phi = 1.618
let dt = 1
let totalSleep = 0
while (totalSleep < timeoutMilli) {
  ...
  dt *= phi
  totalSleep += dt
  yield promiseTimeout(dt)
}
```
This gives the following sleep schedule for a 60s timeout:
```
>>> max = 60000
>>> dt = 1
>>> while total < max:                                                                                   
...   total += dt
...   dt *= 1.618
...   print total
... 
1
2.618
5.235924
9.471725032
16.3252511018
27.4142562827
45.3562666654
74.3864394646
121.357259054
197.356045149
320.322081051
519.28112714
841.196863713
1362.05652549
2204.80745824
3568.37846743
5774.6363603
9344.36163097
15120.1771189
24465.4465784
39586.0925638
64051.2977683
```